provider/hybrid.go

```diff
@@ -0,0 +1,152 @@
+}
+
+// Track a list of native resource tokens
+var dockerImageToken = "docker:index/image:Image"
+
+// gRPC methods for the hybrid provider
+
+func (dp dockerHybridProvider) Attach(ctx context.Context, attach *rpc.PluginAttach) (*emptypb.Empty, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (dp dockerHybridProvider) Call(ctx context.Context, request *rpc.CallRequest) (*rpc.CallResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "Call is not yet implemented")
+}
+
+func (dp dockerHybridProvider) Cancel(ctx context.Context, e *empty.Empty) (*empty.Empty, error) {
+	return &empty.Empty{}, nil
+}
+
+func (dp dockerHybridProvider) GetSchema(ctx context.Context, request *rpc.GetSchemaRequest) (*rpc.GetSchemaResponse, error) {
+	if v := request.GetVersion(); v != 0 {
+		return nil, fmt.Errorf("unsupported schema version %d", v)
+	}
+	return &rpc.GetSchemaResponse{Schema: string(dp.schemaBytes)}, nil
+}
+
+func (dp dockerHybridProvider) CheckConfig(ctx context.Context, request *rpc.CheckRequest) (*rpc.CheckResponse, error) {
+	return &rpc.CheckResponse{Inputs: request.GetNews()}, nil
+}
+
+func (dp dockerHybridProvider) DiffConfig(ctx context.Context, request *rpc.DiffRequest) (*rpc.DiffResponse, error) {
+	return &rpc.DiffResponse{
+		Changes:             0,
+		Replaces:            []string{},
+		Stables:             []string{},
+		DeleteBeforeReplace: false,
+	}, nil
+}
+
+func (dp dockerHybridProvider) Configure(ctx context.Context, request *rpc.ConfigureRequest) (*rpc.ConfigureResponse, error) {
+	var myResp rpc.ConfigureResponse
+	for _, prov := range []rpc.ResourceProviderServer{dp.bridgedProvider, dp.nativeProvider} {
+		resp, err := prov.Configure(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+		myResp = *resp
+	}
+	return &myResp, nil
+}
+
+// TODO: this is for functions AKA data sources, and our provider doesn't have any metadatea becuase we're just implementing from scratch
+func (dp dockerHybridProvider) Invoke(ctx context.Context, request *rpc.InvokeRequest) (*rpc.InvokeResponse, error) {
+	// TODO: remove below snippet, as we're not implementing data sources here atm, or implement a default once the way in which we're passing in any ExtraDataSources is better
+	//if _, ok := dp.metadata.Functions[request.Tok]; ok {
+	//	return dp.nativeProvider.Invoke(ctx, request)
+	//}
+	logging.V(9).Infof("Invoking on bridge provider for: %q", request.Tok)
+	return dp.bridgedProvider.Invoke(ctx, request)
+}
+
+func (dp dockerHybridProvider) StreamInvoke(request *rpc.InvokeRequest, server rpc.ResourceProvider_StreamInvokeServer) error {
+	return status.Error(codes.Unimplemented, "StreamInvoke is not yet implemented")
+}
+
+func (dp dockerHybridProvider) Check(ctx context.Context, request *rpc.CheckRequest) (*rpc.CheckResponse, error) {
+	urn := resource.URN(request.GetUrn())
+
+	tok := urn.Type().String()
+	fmt.Println(tok)
+	// TODO: implement this for actual!!!
+	if tok == dockerImageToken {
```

Comment:
Another interesting pattern when it comes to multiplexing custom resource handlers is what is done in the pulumiservice provider: https://github.com/pulumi/pulumi-pulumiservice/blob/main/provider/pkg/provider/provider.go. Essentially you could add a resource interface which could be used to invoke the handlers for each of the supported resources as done in that provider.